src/sif/autocost.cc

```diff
@@ -408,32 +425,28 @@ Cost AutoCost::EdgeCost(const baldr::DirectedEdge* edge,
   // either the computed edge speed or optional top_speed
   auto edge_speed = tile->GetSpeed(edge, flow_mask_, seconds);
   auto final_speed = std::min(edge_speed, top_speed_);
-
-  float sec = (edge->length() * speedfactor_[final_speed]);
+  float sec = edge->length() * speedfactor_[final_speed];
 
   if (shortest_) {
     return Cost(edge->length(), sec);
   }
 
-  float factor =
-      (edge->use() == Use::kFerry)
-          ? ferry_factor_
-          : (edge->use() == Use::kRailFerry) ? rail_ferry_factor_ : density_factor_[edge->density()];
+  // base factor is either ferry, rail ferry or density based
+  float factor = (edge->use() == Use::kFerry) * ferry_factor_ +
+                 (edge->use() == Use::kRailFerry) * rail_ferry_factor_ +
+                 (edge->use() != Use::kFerry && edge->use() != Use::kRailFerry) *
+                     density_factor_[edge->density()];
 
-  // TODO: factor hasn't been extensively tested, might alter this in future
+  // TODO: speed_penality hasn't been extensively tested, might alter this in future
   float speed_penalty = (edge_speed > top_speed_) ? (edge_speed - top_speed_) * 0.05f : 0.0f;
-  factor += highway_factor_ * kHighwayFactor[static_cast<uint32_t>(edge->classification())] +
-            surface_factor_ * kSurfaceFactor[static_cast<uint32_t>(edge->surface())] + speed_penalty;
-
-  if (edge->toll()) {
-    factor += toll_factor_;
-  }
-
-  if (edge->use() == Use::kAlley) {
-    factor *= alley_factor_;
-  }
-
-  return Cost(sec * factor, sec);
+  factor += (highway_factor_ * kHighwayFactor[static_cast<uint32_t>(edge->classification())] +
+             surface_factor_ * kSurfaceFactor[static_cast<uint32_t>(edge->surface())] +
+             speed_penalty + edge->toll() * toll_factor_) *
+            (edge->use() == Use::kAlley ? alley_factor_ : 1.f);
+
+  // base cost before the factor is a linear combination of time vs distance, depending on which one
+  // the user thinks is more important to them
+  return Cost((sec * inv_distance_factor_ + edge->length() * distance_factor_) * factor, sec);
```

Comment:
here is where we use a linear combination for the edges time vs the edges length